File: frontend/app-autor/src/app/services/libro.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Libro } from '../models/Libro';
import { Observable } from 'rxjs';
import { LibroDTO } from '../models/LibroDTO';

@Injectable({
  providedIn: 'root',
})
export class LibroService {
  private api = 'http://localhost:8080/api/libro';

  constructor(private http: HttpClient) {}

  getLibros(): Observable<Libro[]> {
    return this.http.get<Libro[]>(this.api+'/getLibros');
  }

  updateLibro(id: number, libro: LibroDTO): Observable<any> {
    return this.http.put<Libro>(this.api+'/updateLibro/'+id, libro);
  }

  deleteLibro(id: number){
    return this.http.delete<Libro>(this.api+'/deleteLibro/'+id);
  }

  addLibro(libro: Libro, dni: number){
    return this.http.post<Libro>(this.api+'/saveLibro/'+dni, libro);
  }

}

File: backend/scaffolding-cc-main/src/main/java/scaffolding/cc/services/LibroService.java
package scaffolding.cc.services;

import org.springframework.stereotype.Service;
import scaffolding.cc.dto.LibroDTO;
import scaffolding.cc.entities.LibroEntity;
import scaffolding.cc.models.Libro;

import java.util.List;

@Service
public interface LibroService {
    List<Libro> getLibros();

    Libro getLibroById(Long id);

    LibroEntity saveLibro(LibroDTO libro, Integer dni);

    Boolean deleteLibro(Long id);

    LibroEntity updateLibro(Long id, LibroDTO libro);

    List<Libro> getLibrosByAutor(Integer dni);
}
